### ISSUE TITLE
Remove Google Analytics

### ISSUE THREAD

**Issue by KaptajnKold** [h5bp/html5-boilerplate#1275]
It concerns me that H5BP promotes the use of Google Analytics as a best practice by including the GA snippet in the boilerplate. There a privacy issues to be considered when using GA and I for one am not comfortable with the fact that more than 50% of all websites are sending information about their visitors to the same third party. The ubiquity of GA gives Google the ability to track individuals browsing from site to site and while the Analytics tools themselves are nifty for web masters wanting to optimize their sites, they come at a privacy cost for their users.

In short, I think the snippet should be removed lest we want web masters to mindlessly infringe on their users privacy because they think it's somehow best practice. Let it be up to the web masters to consciously choose which tracking tools if any they want to employ.

**Comment by guilhermesimoes** [comment]
Funny how just 6 months after this issue was closed we got the Snowden revelations. I think @KaptajnKold's arguments are even stronger today than they were 3 years ago.

> you haven't provided a strong argument about the privacy risks either.

Is this enough? [NSA uses Google cookies to pinpoint targets for hacking, the Washington Post reports](http://www.washingtonpost.com/blogs/the-switch/wp/2013/12/10/nsa-uses-google-cookies-to-pinpoint-targets-for-hacking/).

I think this issue could be reopened so we could have an honest discussion about the implications of using Google Analytics.

**Comment by Volker-E** [comment]
I agree strongly with @KaptajnKold argument on the popularity of HTML5 Boilerplate -- that it makes beginners think what's the standard/it sets the standards without them necessarily asking for the why. 
Why explicitly integrating Google Analytics and not Piwik code for your own server? 
For me that should be more a side-fork like different server configurations of H5BP.

**Comment by roblarsen** [comment]
>Why explicitly integrating Google Analytics and not Piwik code for your own server? 

*Potential privacy issues aside*, the two technologies aren't even in the same conversation when it comes to usage. Offering GA as part of the base HTML5  Boilerplate package serves a majority- including many millions of people who don't have access to their own server to install their own analytics package. The HTML5 Boilerplate project selling Piwik to people is just a crazy idea especially since many (most?) developers aren't the ones actually making the decision about what analytics package to use,

http://trends.builtwith.com/analytics/Google-Analytics > 50% of the web

http://trends.builtwith.com/analytics/Piwik-Web-Analytics > 0.5% of the web

**Comment by Volker-E** [comment]
Look, and here we are at the Facebook-Like-Button again. I think, that none of those should be unquestioned default.

**Comment by roblarsen** [comment]
I don't understand how the facebook like button keeps coming into this conversation.

**Comment by guilhermesimoes** [comment]
> Potential privacy issues aside

First of all, they're not *potential*, they're **real** privacy issues.

Second of all, why should these issues be put aside? As developers our main concern should always be our users, shouldn't we do our very best to fight for their privacy? I'm certain most users are not even aware of what Google Analytics is and what it does.

**Comment by patrickkettner** [comment]
> As developers our main concern should always be our users

sure. so delete the non working snippet. its a boilerplate, man.

The issue here is whether or not a collection of best practices for making a website should include a non working version of something that a huge percentage of the internet uses.

No one is forcing anyone to use or do anything. Its an opt in feature of an opt in boilerplate.

**Comment by roblarsen** [comment]
> Second of all, why should these issues be put aside?

Some people involved in this discussion are very passionate about this issue. Therefore, if I want to make a separate point about usage,and how piwik is a non-starter, I need to at least attempt to separate out the privacy issue because that otherwise that passion overwhelms my point. Which it's doing anyway.

**Comment by roblarsen** [comment]
To add onto my stance on this from a few years ago

>The reality is, whether you like it or not, it's extraordinarily common for people to include an analytics package. Many more than would use Backbone or jQuery UI. I've worked on dozens of commercial sites and I'd have to go way back in time to find one that doesn't have some sort of analytics.

>This project could either make a stand on a privacy issue or could help developers with a very common task. I lean towards helping people get their jobs done.

Adding to that, if we decided that we wanted to make a stand about privacy, I would prefer to have no analytics code at all rather than shipping something no one is going to use. If we ship with some project no one has ever heard of instead of GA we're just getting in the way of developers when they inevitably install GA or Omniture or whatever their product owner/project manager/ client wants. 

This has never been a choice I've made in 10 years as a consultant. I've been asked about different options, but the people that pay my bills are the people making decisions about analytics software. It's simply not up to me. It'd be like me telling designers to use Gimp instead of Photoshop.

**Comment by guilhermesimoes** [comment]
Only Volker-E is arguing for Piwik (I don't even know what it is). That discussion should be carried on on issue  #1547.

> If we decided that we wanted to make a stand about privacy, I would prefer to have no analytics code at all rather than shipping something no one is going to use

@roblarsen We're in agreement then!

> No one is forcing anyone to use or do anything. Its an opt in feature of an opt in boilerplate.

@patrickkettner As the OP said, by including GA in the boilerplate you are is advocating for its use. Opt-in features are found in the [Extend docs][1], we could move this opt-in feature to exactly that place.

[1]: https://github.com/h5bp/html5-boilerplate/blob/master/dist/doc/extend.md#google-universal-analytics

**Comment by patrickkettner** [comment]
> by including GA in the boilerplate you are is advocating for its use. 

I would disagree. Up until recently included code to support IE7, but certainly were not advocating for its use. Boilerplate should be reactionary. When something is extremely popular, support should be included.

**Comment by guilhermesimoes** [comment]
> Up until recently included code to support IE7, but certainly were not advocating for its use.

Your comparison with IE7 is not accurate. By including code to support IE7, you're advocating for the support of older browsers like IE7. By including GA, you're advocating for its use in tracking users.

> When something is extremely popular, support should be included.

Most people on the internet use either Twitter or Facebook (or both) so it's obvious you'd want a link to your site to look good on those platforms, right? And yet Twitter Cards and Facebook Open Graph Data are [not included by default][2] in the boilerplate.

> Boilerplate should be reactionary.

How about reacting to [this][2]? :)

[1]: https://github.com/h5bp/html5-boilerplate/blob/master/dist/doc/extend.md#social-networks
[2]: http://apps.washingtonpost.com/g/page/national/nsa-signal-surveillance-success-stories/647/#document/p3/a135602

**Comment by patrickkettner** [comment]
> you're advocating for the support of older browsers like IE7. 

no. no we are not. we are saying that they are popular enough to warrant adding support for.

> Most people on the internet use either Twitter or Facebook (or both) so it's obvious you'd want a link to your site to look good on those platforms, right? 

no, most websites do not use those things. a large amount do, but nowhere near the amount that use google analytics.

**Comment by Volker-E** [comment]
I'm ***not*** for Piwik as default, my question above was meant rhetorically and seemed misleading. **I'm agreeing** with commenters before me -- I question shipping GA as always included default.

Getting back to @roblarsen who was thankfully sharing the insightful link with data on usage of GA (50-60% in the top sites and 9.2% sites of the entire Internet) -- that's impressive, still doesn't convince me.
The before mentioned [Facebook Like Widget](http://trends.builtwith.com/widgets/Facebook-Like) (most popular, there are [a few different](http://builtwith.com/?q=facebook+like)) has got 20-26.5% coverage in top sites and 4.4% sites of the entire Internet according to builtwith.com

Please also compare numbers for [HTML5 Doctype](http://trends.builtwith.com/docinfo/HTML5-DocType) and [HTML5Boilerplate](http://trends.builtwith.com/docinfo/HTML5-Boilerplate)
